core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.spec.ts

```diff
@@ -141,6 +167,315 @@ describe('ExplorationFooterComponent', () => {
+        then: (successCallback: () => void, errorCallback: () => void) => {
+          successCallback();
+          errorCallback();
+        }
+      }
+    } as NgbModalRef);
+
+    component.openInformationCardModal();
+
+    expect(ngbModal.open).toHaveBeenCalled();
+    expect(component.checkpointArray).toEqual([0, 1]);
+  });
+
+  it('should push 0 in checkpointArray when card_is_checkpoint' +
+    'is false and displayedCardIndex > 1', () => {
+    let ngbModal = TestBed.inject(NgbModal);
+
+    expect(component.checkpointArray).toEqual([0]);
+    let stateName = 'stateName';
+    let state =
+      {
+        classifier_model_id: null,
+        content: {
+          html: '',
+          content_id: 'content'
+        },
+        interaction: {
+          id: 'FractionInput',
+          customization_args: {
+            requireSimplestForm: { value: false },
+            allowImproperFraction: { value: true },
+            allowNonzeroIntegerPart: { value: true },
+            customPlaceholder: { value: {
+              content_id: '',
+              unicode_str: ''
+            } },
+          },
+          answer_groups: [],
+          default_outcome: {
+            dest: 'Introduction',
+            feedback: {
+              content_id: 'default_outcome',
+              html: ''
+            },
+            labelled_as_correct: false,
+            param_changes: [],
+            refresher_exploration_id: null,
+            missing_prerequisite_skill_id: null
+          },
+          confirmed_unclassified_answers: [],
+          hints: [],
+          solution: null
+        },
+        linked_skill_id: null,
+        next_content_id_index: 0,
+        param_changes: [],
+        recorded_voiceovers: {
+          voiceovers_mapping: {
+            content: {},
+            default_outcome: {}
+          }
+        },
+        solicit_answer_details: false,
+        card_is_checkpoint: false,
+        written_translations: {
+          translations_mapping: {
+            content: {},
+            default_outcome: {}
+          }
+        }
+      };
+
+    spyOn(playerPositionService, 'getDisplayedCardIndex').and.
+      returnValue(1);
+
+    spyOn(explorationEngineService, 'getState').and.returnValue(
+      stateObjectFactory.createFromBackendDict(stateName, state));
+
+    spyOn(ngbModal, 'open').and.returnValue({
+      componentInstance: {
+        numberofCheckpoints: 0,
+        completedWidth: 0,
+        contributorNames: [],
+        expInfo: null
+      },
+      result: {
+        then: (successCallback: () => void, errorCallback: () => void) => {
+          successCallback();
+          errorCallback();
+        }
+      }
+    } as NgbModalRef);
+
+    component.openInformationCardModal();
+
+    expect(ngbModal.open).toHaveBeenCalled();
+    expect(component.checkpointArray).toEqual([0, 0]);
+  });
+
+  it('should push 1 in checkpointArray when card_is_checkpoint' +
+    'is true and displayedCardIndex > 1', () => {
```

Comment:
Ditto: rename description so it does not refer to variable names and implementation specifics